### PR TITLE
Fix/tooltip

### PR DESCRIPTION
**Pull request by diegonzs-roll**
## What's done
Use FloatingPortal to render the tooltip
## Tasks

- [ ] Have you written tests for new code (if applicable)?
- [ ] Have you tested the changes on all the platforms?
  - [ ] iOS
  - [ ] Android
  - [x] Web
- [ ] Have you added stories to demonstrate the new functionality (if there is any)?

### CODE REVIEW

File: packages/design-system/src/atoms/tooltip/index.web.tsx [hunk 1]
```diff
@@ -66,28 +67,30 @@ export const Tooltip: React.FC<TooltipProps> = ({
...
-              position: strategy,
-            },
-          ]}
-          ref={floating}
-          {...getFloatingProps()}
-        >
-          {asTextNode(title, variant === 'dark' ? white : charcoalBlack)}
-        </View>
-      )}
+      <FloatingPortal>
```
Comment (teimurjan) [inline packages/design-system/src/atoms/tooltip/index.web.tsx:70]: Let's make it configurable. Let's define a boolean prop `renderInPortal` and use `FloatingPortal` depending on its value.

File: packages/design-system/src/atoms/conditionalWrapper/index.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { FC, Fragment } from 'react'
+
+export interface ConditionalWrapperProps {
+  condition: boolean
+  wrapper: (children: React.ReactNode) => JSX.Element
+}
+
+export const ConditionalWrapper: FC<ConditionalWrapperProps> = ({
```
Comment (diegonzs-roll) [inline packages/design-system/src/atoms/conditionalWrapper/index.tsx:8]: Added ConditionalWrapper as an atom so we can reuse it in other projects.

File: packages/design-system/src/atoms/tooltip/index.web.tsx [hunk 3]
```diff
@@ -61,33 +64,39 @@ export const Tooltip: React.FC<TooltipProps> = ({
...
-              position: strategy,
-            },
-          ]}
-          ref={floating}
-          {...getFloatingProps()}
-        >
-          {asTextNode(title, variant === 'dark' ? white : charcoalBlack)}
-        </View>
-      )}
+      <ConditionalWrapper condition={renderInPortal} wrapper={wrapper}>
```
Comment (diegonzs-roll) [inline packages/design-system/src/atoms/tooltip/index.web.tsx:76]: Now  we use the prop renderInPortal and the ConditionalWrapper component to render the tooltip inside the FloatingPortal component.

File: .changeset/ninety-windows-look.md [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+'@tryrolljs/design-system': patch
+---
+
+Use FloatingPortal to render tooltip
+Add ConditionalWrapper atom component
+Add boolean prop on tooltip component to wrap it with FloatingPortal
```
Comment (teimurjan) [inline .changeset/ninety-windows-look.md:7]:
```suggestion
- Use FloatingPortal to render tooltip
- Add ConditionalWrapper atom component
- Add boolean prop on tooltip component to wrap it with FloatingPortal
```

File: packages/design-system/src/atoms/tooltip/index.web.tsx [hunk 5]
```diff
@@ -61,33 +64,39 @@ export const Tooltip: React.FC<TooltipProps> = ({
     useFocus(context),
   ])
 
+  const wrapper = useCallback((render: React.ReactNode) => {
+    return <FloatingPortal>{render}</FloatingPortal>
+  }, [])
+
```
Comment (teimurjan) [inline packages/design-system/src/atoms/tooltip/index.web.tsx:70]: You can declare it outside of the component context, so we save memory (no `useCallback` => no memoization)

**Review by teimurjan — APPROVED** [review]
Approved, but I'd like my comments to be addressed before the merge.

**Pull request merged**